packages/flutter/lib/src/rendering/paragraph.dart

```diff
@@ -225,10 +223,7 @@ class RenderParagraph extends RenderBox {
     final Canvas canvas = context.canvas;
     if (_hasVisualOverflow) {
       final Rect bounds = offset & size;
-      if (_overflowPainter != null)
-        canvas.saveLayer(bounds, new Paint());
-      else
-        canvas.save();
+      canvas.save();
```

Comment:
I think we still need to do a saveLayer here if we need to generate the fade effect because the fade effect needs to blend the text against the background.